### PR TITLE
ncm-opennebula:  Include support to tune ONE VMM drivers

### LINKED ISSUE #584: ncm-opennebula: Include support to tune ONE VMM drivers
It's possible to tune some KVM/XEN options from ONE VMM drivers. Include support for these options into OpenNebula component:

http://docs.opennebula.org/4.12/administration/virtualization/vmmg.html

### PR DESCRIPTION
**Pull request by alvarosimon**
This PR includes kvmrc file support to tune libvirt/qemu params on each hyp.
**NOTE:** This PR depends on #572 which must be merged in first place.

 * Fix #584: Include support to tune ONE VMM drivers
 * Include sunstone `remote` auth (ONE 4.14.2)

### CODE REVIEW

File: ncm-opennebula/src/main/perl/opennebula.pm [hunk 1]
```diff
@@ -325,6 +330,19 @@ sub change_opennebula_passwd
...
 
+# Sync hyps VMMs scripts
+sub sync_opennebula_hyps
+{
+    my ($self, $cmd) = @_;
+    my $output;
+
+    $output = $self->run_onehost_as_oneadmin_with_ssh($cmd, "localhost", 0);
+    if (!$output) {
+        $self->error("Quattor unable to execute onehost command.");
```
Comment (stdweird) [inline ncm-opennebula/src/main/perl/opennebula.pm]: error message should mention what was trying to do (likethe info message does), not only what failed.

File: ncm-opennebula/src/main/perl/opennebula.pm [hunk 2]
```diff
@@ -334,8 +352,11 @@ sub restart_opennebula_service {
         $srv = CAF::Service->new(['opennebula'], log => $self);
     } elsif ($service eq "sunstone") {
         $srv = CAF::Service->new(['opennebula-sunstone'], log => $self);
+    } elsif ($service eq "kvmrc") {
+        $self->info("Updated $service file. onehost sync is required.");
+        $self->sync_opennebula_hyps("-f");
```
Comment (stdweird) [inline ncm-opennebula/src/main/perl/opennebula.pm]: what does the `-f` mean?

**Comment by stdweird** [comment]
@alvarosimon some minor remarks, looks good to be merged.

File: ncm-opennebula/src/main/perl/opennebula.pm (hunk 2 above)
Comment (alvarosimon) [inline ncm-opennebula/src/main/perl/opennebula.pm]: it forces to update VMM drivers on each hyp. Just to be sure that all hyps are updated if VMM conf is changed by quattor.

Comment (stdweird) [inline ncm-opennebula/src/main/perl/opennebula.pm]: can you add a comment? passing the `-f` as `cmd` just looks very strange

Comment (alvarosimon) [inline ncm-opennebula/src/main/perl/opennebula.pm]: @stdweird maybe is better to include it into `@ONEHOST_SYNC_COMMAND` within commands.pm as default value:

> Readonly::Array my @ONEHOST_SYNC_COMMAND => qw(/usr/bin/onehost sync -f);

Instead include `-f` option as `cmd`, what do you think?

Comment (stdweird) [inline ncm-opennebula/src/main/perl/opennebula.pm]: yes, sounds good

Comment (alvarosimon) [inline ncm-opennebula/src/main/perl/opennebula.pm]: Done! now should be more clear...

**Comment by jrha** [comment]
test this please

**Comment by stdweird** [comment]
LGTM

**Comment by jrha** [comment]
@apdibbo, you may want to have a look at this one too.

**Comment by jrha** [comment]
@alvarosimon #572 is merged, can you rebase this one please?

**Comment by alvarosimon** [comment]
@jrha Done!

**Pull request merged**